library/Freckle/App/Database.hs

```diff
@@ -76,15 +78,15 @@ import Yesod.Core.Types (HandlerData (..), RunHandlerEnv (..))
 
 -- | A monadic context in which a SQL backend is available
 --   for running database queries
-class MonadIO m => MonadSqlBackend m where
+class MonadUnliftIO m => MonadSqlBackend m where
   getSqlBackendM :: m SqlBackend
```

Comment:
This is actually a little surprising to me. I once mentioned that `MonadTx` (or whatever it is) would make sense with a `MonadUnliftIO` constraint because it's implicitly bracket-like, but this class has no reason to implicitly want to do `MonadUnliftIO` stuff, IMO. I would've probably not even had it be `MonadIO` but only `Monad`. 

So, it would make sense to me to leave this as-is (or make it `Monad` only) and then have `(MonadUnliftIO m, MonadSqlBackend m)` on `liftSql` instead. But doesn't really matter either way, I guess.

Reply:
I would rather have it be `MonadCatch` for what it's worth.

Reply:
Alternatively... you can always generalize further. New typeclass to abstract over annotated exception handling.